src/main/scala/com/databricks/labs/overwatch/pipeline/GoldTransforms.scala

```diff
@@ -493,6 +493,82 @@ trait GoldTransforms extends SparkSessionWrapper {
 
   }
 
+
+  protected def buildNotebookCommands(
+                                   notebook: PipelineTable,
+                                   clsfIncrementalDF : DataFrame,
+                                 )(auditIncrementalDF: DataFrame): DataFrame = {
+
+
+    val notebookLookupTSDF = notebook.asDF
+      .select("organization_id", "notebook_id", "notebook_path", "notebook_name", "unixTimeMS","date")
+      .withColumnRenamed("notebook_id", "notebookId")
+      .filter('notebookId.isNotNull)
+      .distinct
+      .toTSDF("unixTimeMS", "organization_id","date","notebookId")
+
+    val clsfLookupTSDF = clsfIncrementalDF
+      .select(
+        "organization_id", "state_start_date", "unixTimeMS_state_start", "cluster_id", "cluster_name",
+        "custom_tags", "node_type_id", "current_num_workers", "uptime_in_state_H", "total_cost")
+      .distinct
+      .withColumnRenamed("unixTimeMS_state_start", "unixTimeMS")
+      .withColumnRenamed("cluster_id", "clusterId")
+      .withColumnRenamed("current_num_workers", "node_count")
+      .withColumn("totalCostPMS", col("total_cost") / col("uptime_in_state_H")/lit(3600))
+      .toTSDF("unixTimeMS", "organization_id", "clusterId")
+
+    val colNames: Array[Column] = Array(
+      'organization_id,
+      'clusterId.alias("cluster_id"),
+      'notebookId.alias("notebook_id"),
+      'workspace_name,
+      'date,
+      'timestamp,
+      'notebook_path,
+      'notebook_name,
+      'commandId.alias("command_id"),
+      'commandText.alias("command_text"),
+      'executionTime.alias("execution_time_s"),
+      'sourceIpAddress.alias("source_ip_address"),
+      'userIdentity.alias("user_identity"),
+      'estimated_dbu_cost,
```

Comment:
In your proof you should add a sanity check for this. Find a command that ran for some reasonable amount of time (longer is better to get bigger numbers), then do a manual calculation of how much you think it should cost based on the time cost of that clusters state while it ran and make sure what you're getting for this field is fairly close to it. I don't think this calc is right yet.

Reply:
reminder to do this sanity check ^